sql/expression/function/regexp_like.go

```diff
@@ -209,29 +235,43 @@ func compileRegex(ctx *sql.Context, pattern, flags sql.Expression, funcName stri
 		if err != nil {
 			return nil, err
 		}
-		flagsStr = fmt.Sprintf("(?%s)", flagsStr)
-		flagsStr = strings.Replace(flagsStr, "c", `\c`, -1)
 	}
-	return regexp.Compile(flagsStr + patternVal.(string))
+	regexFlags := regex.RegexFlags_None
+	for _, flag := range flagsStr {
+		// The 'c' flag is the default behavior, so we don't need to set anything in that case.
+		// Any illegal flags will have been caught by consolidateRegexpFlags.
+		switch flag {
+		case 'i':
+			regexFlags |= regex.RegexFlags_Case_Insensitive
+		case 'm':
+			regexFlags |= regex.RegexFlags_Multiline
+		case 'n':
+			regexFlags |= regex.RegexFlags_Dot_All
+		case 'u':
+			regexFlags |= regex.RegexFlags_Unix_Lines
+		}
+	}
+
+	re := regex.CreateRegex()
+	return re, re.SetRegexString(ctx, patternVal.(string), regexFlags)
```

Comment:
This is a bit awkward to use and finalize as written here, and I think there are some leaks as a result. Clearer/safer would be:

```go
if err := re.SetRegexString(ctx, patternVal.(string), regexFlags); err != nil {
    re.Close(ctx)
    return err
}
return re, nil
```